[Scala] new concept exercise operators (#2205)
exercism/v3: languages/scala/exercises/concept/operators/.docs/instructions.md

In this exercise you're going to write a calculator for rectangles.

Given is a case class `Rectangle`. A rectangle is defined by its width and its height:

```scala
case class Rectangle(width: Int, height: Int)
```

> The calculator will allow to sum up two rectangles and to increase a rectangle by multiplying it with a number.

Reviewer: ```suggestion
The calculator will allow to sum two rectangles and to expand a rectangle by multiplying it with a number.
```

> ## 1. Create a plus operator for rectangles

Reviewer: This heading is a bit technical. As per [the docs](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsinstructionsmd):

> The heading should describe what to implement, not how to implement it 

Maybe something like:

```suggestion
## 1. Allow rectangles of the same height to be added
```

Or something like that.

Reviewer 2: Thanks, good general hint.

Implement a `+` method that takes another rectangle as argument. Assume that the passed rectangle has the same height as the current one. The method should return a new rectangle, whose width is a sum of both rectangle's widths and whose height is the same as of both rectangles.

Let's visualize the expected outcome:

> ![sum of 2 rectangles of same height](sum_same_height.png)

Reviewer: This exercise uses an image to make things clearer, which we usually don't do in exercise. There are several reasons for this, including images not always being suitable for different form factors. @iHiD have we made a decision on using images in exercises?

I think that in this case, the description together with the example might be good enough to allow not using an image.

Reviewer 3: Yeah - we can't have images as these need to work when you download it and read it just as markdown. Also it would break accessibility if the image was critical to the exercise.

Reviewer 2: Ah, what a pity. Ok fine, I will throw them out.

> This code snippet shows how to use the `+` method:

Reviewer: I think you can omit this.

```scala
val leftRectangle = Rectangle(2, 5)
val rightRectangle = Rectangle(1, 5)
val sumRectangle = leftRectangle + rightRectangle
println(sumRectangle)
// => Rectangle(3, 5)
```

> ## 2. Enhance the plus operator

Reviewer: Similar to the comment on previous task, this could maybe be phrased more in terms of _what_ to do instead instead of _how_. Maybe something like:

```suggestion
## 2. Allow adding rectangles of different heights
```

How to deal with rectangles of different heights? In this task please update the `+` method like follows: The new rectangle should get the height of the higher rectangle.

Let's visualize again the expected outcome:

> ![sum of 2 rectangles of same height](sum_different-height.png)

Reviewer: See previous comment on using images.

> This is a code example shows again the expected result:

Reviewer: I think you can omit this.

```scala
val leftRectangle = Rectangle(2, 4)
val rightRectangle = Rectangle(1, 1)
val sumRectangle = leftRectangle + rightRectangle
println(sumRectangle)
// => Rectangle(3, 4)
```

> ## 3. Create a multiply operator for rectangles

Reviewer: See previous task heading comments. Maybe something like:

```suggestion
## 3. Allow multiplying rectangles by a factor
```

> Implement a `*` method, which takes an Int parameter and returns an expanded rectangle, whose width and height are multiplied by the given parameter.

Reviewer: ```suggestion
Implement a `*` method, which takes an `Int` parameter and returns an expanded rectangle, whose width and height are multiplied by the given parameter.
```

> ![multiplication of a rectangle with a number](multiplication_rectangle_number.png)

Reviewer: See previous comment on using images.

This code snippet shows the usage of the new `*` method.

> 

Reviewer: I think you can omit this.
